### ISSUE TITLE
About pre-trained model

### ISSUE THREAD

**Issue by socoi** [ignaciorlando/red-lesion-detection#9]
Hi. I am interested in your work.
I design a CNN model to extract features and plan to combine your hand-crafted features.
I wonder whether you may give an explanation about your
pre-trained model? Should be different 63-dimension features vector for the whole training sets?

Thanks a lot.

**Comment by ignaciorlando** [comment]
Hi @socoi, thanks for your interest!
Sure, I'm happy to help. If you want to extract the hand crafted features on your own data, you need to call the `hand_crafted_features_extraction` (https://github.com/ignaciorlando/red-lesion-detection/blob/master/RedLesionDetection/util/feature_extraction/hand_crafted_features_extraction.m).
Let me know if you have any other question!
Cheers,

Nacho

**Comment by socoi** [comment]
@ignaciorlando thanks a lot.
I am trying to re-build code using Python. Since fundus images may extract a different number of candidates. I could extract 83 candidates and get [83,63] for an image. If I want to solely use this feature for training a random forest, how can I know each candidate belongs to which class? DR or not?
Thank you.

**Comment by ignaciorlando** [comment]
Hi @socoi,

Let's use an example. Imagine that you have 2 images, I_1 and I_2, each of them with N_1 and N_2 candidates. From each of these candidates you can extract a d-dimensional feature vector, and build a design matrix X of size [N_1  + N_2, d] by concatenating each sample.

For each image you also have the ground truths Y_1 and Y_2, which are binary images with the same size of the images, with True representing true lesions and False representing any other structure. We considered a candidate as a True positive if it overlaps with a True label at least in one pixel. Then, you can get a vector of labels y by analyzing the overlapping between each candidate and the ground truth. Finally, you can use (X, y) to train a classifier.

Please, take into account that the classifier will be able to determine if a given candidate is a red lesion or not. You can use the probabilities of each candidates as a proxy or a score for DR detection. In order to do that, you can follow our approach of the paper, which was to take the maximum probability of all the candidates in a given image. That will give you a probability p_1 and p_2 for the two example images indicating if it might be associated to a DR patient or not.

Hope it helps! Sorry for the delay, I've been really busy with other stuff.
